clients/src/main/java/org/apache/kafka/common/network/Selector.java

```diff
@@ -1072,6 +1067,52 @@ public int numStagedReceives(KafkaChannel channel) {
         return deque == null ? 0 : deque.size();
     }
 
+    class SelectorChannelMetadataRegistry implements ChannelMetadataRegistry {
+        private CipherInformation cipherInformation;
+        private ClientInformation clientInformation;
+
+        @Override
+        public void registerCipherInformation(final CipherInformation cipherInformation) {
+            if (this.cipherInformation == null) {
+                this.cipherInformation = cipherInformation;
+                sensors.connectionsByCipher.increment(cipherInformation);
+            }
+        }
+
+        @Override
+        public CipherInformation cipherInformation() {
+            return cipherInformation;
+        }
+
+        @Override
+        public void registerClientInformation(final ClientInformation clientInformation) {
+            if (this.clientInformation != null) {
```

Comment:
If this already exists, we should check to see if it's equal to the new information we wanted to register.  If it is, we can skip the decrement metric + increment metric dance.

Reply:
Ack. That makes sense.